src/index.ts

```diff
@@ -215,35 +225,57 @@ export default class Kumulos {
         return trackEvent(this.context, type, properties).then(_ => void 0);
     }
 
+    async getPushSubscriptionStatus(
+        subscriptionStateListener: (
+            subscriptionState: PushSubscriptionState
+        ) => void
+    ) {
+        const subscriptionState = await this.pushManager.getCurrentSubscriptionState(
+            this.context
+        );
+
+        subscriptionStateListener(subscriptionState);
+    }
+
     async pushRegister(): Promise<void> {
-        const pushManager = await getPushOpsManager(this.context);
-        const permission = await pushManager.requestNotificationPermission(
+        const permission = await this.pushManager.requestNotificationPermission(
             this.context
         );
 
         if (permission !== 'granted') {
             return Promise.reject('Notification permission not granted');
         }
 
-        //TODO: The below code is a hack in place to avoid an issue with the onPermissionChange event not firing from Safari: https://bugs.webkit.org/show_bug.cgi?id=256201#c1
-        // it also supports the legacy safari push notification implementation, which can not rely on the permission change event handler
-        const browser = getBrowserName();
+        this.pushManager.pushRegister(this.context);
+    }
 
-        if (browser === 'safari' || !navigator.permissions) {
-            pushManager.pushRegister(this.context);
-        }
```

Comment:
Was this logic moved somewhere else / is it safe to remove?

Reply:
We are calling register each time now, even if not 'safari', previously we counted on the permission observer to finish the registration, now we need to know specifically that it was from an API call

Reply:
This will result in 2 calls to pushregister instead of one, which was covered in [this pr](https://github.com/Kumulos/KumulosSdkWeb/pull/62/files) which came after the permission observer